spec/unit/parser/functions/sprintf_spec.rb

```diff
@@ -44,4 +44,25 @@
     expect(result).to(eql("<overlong:  027   0XBEEF (foo     )>"))
   end
 
+  it 'does not attempt to mutate its arguments' do
+    args = ['%d', 1].freeze
+    expect { @scope.function_sprintf(args) }.to_not raise_error
+  end
+
+  it 'support named arguments in a hash with string keys' do
+    result = @scope.function_sprintf(["%<foo>d : %<bar>f", {'foo' => 1, 'bar' => 2}])
+    expect(result).to eq("1 : 2.000000")
+  end
+
+  it 'raises a key error if a key is not present' do
+    expect do
+      @scope.function_sprintf(["%<foo>d : %<zanzibar>f", {'foo' => 1, 'bar' => 2}])
+    end.to raise_error(KeyError, /key<zanzibar> not found/)
+  end
+
+  it 'a hash with string keys that is output formats as strings' do
+    result = @scope.function_sprintf(["%s", {'foo' => 1, 'bar' => 2}])
+    expect(result).to eq("{\"foo\"=>1, \"bar\"=>2}")
+  end
+
 end
```

Comment:
Should test that arbitrary arguments are allowed and ignored. I.e. no attempt is made to make symbols out of integers and no errors are raised for entries that doesn't match.